test/e2e_node/util.go

```diff
@@ -418,18 +418,34 @@ func getCRIClient() (internalapi.RuntimeService, internalapi.ImageManagerService
 }
 
 // TODO: Find a uniform way to deal with systemctl/initctl/service operations. #34494
-func restartKubelet() {
+func findRunningKubletServiceName() string {
 	stdout, err := exec.Command("sudo", "systemctl", "list-units", "kubelet*", "--state=running").CombinedOutput()
 	framework.ExpectNoError(err)
 	regex := regexp.MustCompile("(kubelet-\\w+)")
 	matches := regex.FindStringSubmatch(string(stdout))
 	framework.ExpectNotEqual(len(matches), 0)
-	kube := matches[0]
-	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kube)
-	stdout, err = exec.Command("sudo", "systemctl", "restart", kube).CombinedOutput()
+	kubeletServiceName := matches[0]
+	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kubeletServiceName)
+	return kubeletServiceName
+}
+
+func restartKubelet() {
+	kubeletServiceName := findRunningKubletServiceName()
+	stdout, err := exec.Command("sudo", "systemctl", "restart", kubeletServiceName).CombinedOutput()
 	framework.ExpectNoError(err, "Failed to restart kubelet with systemctl: %v, %v", err, stdout)
 }
 
+// stopKubelet will kill the running kubelet, and returns a func that will restart the process again
+func stopKubelet() func() {
+	kubeletServiceName := findRunningKubletServiceName()
+	stdout, err := exec.Command("sudo", "systemctl", "kill", kubeletServiceName).CombinedOutput()
```

Comment:
I'd suggest to use ```systemctl stop``` here. Unlike ```systemctl kill``` it will use official configured way to stop the service.

Reply:
When using `systemctl stop` the systemd service will be deleted, making it impossible to start it again.

> If killing kubelet is mandatory for your tests then it would be better to call this function killKubelet.

Not sure what you mean with `killKubelet`.

Reply:
> When using systemctl stop the systemd service will be deleted, making it impossible to start it again.

Can you elaborate on this? Not sure I follow.

> Not sure what you mean with killKubelet.

I mean that you called function stopKubelet, but it doesn't stop Kubelet, it kills it. I proposed to call it KillKubelet.

Reply:
> Can you elaborate on this? Not sure I follow.

```bash
# Using stop
> systemd-run --remain-after-exit sleep 100
Running as unit: run-u14192.service
> systemctl stop run-u14192.service
> systemctl start run-u14192.service
Failed to start run-u14192.service: Unit run-u14192.service not found.

# Using kill                                                                                                                                                                                             
> systemd-run --remain-after-exit sleep 100
Running as unit: run-u14224.service 
> systemctl kill run-u14224.service
> systemctl start run-u14224.service
# service is restarted
```

> I mean that you called function stopKubelet, but it doesn't stop Kubelet, it kills it. I proposed to call it KillKubelet.

Aha, I agree that renaming it would make sense. I thought you meant to call a given function named `killKubelet` instead.

Reply:
we can rename the function in a follow-on, but i don't see a major reason to block on the core issue addressed in the pr.